### PR TITLE
address: removing the throwing variant of parseInternetAddress

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: updated tests
Docs Changes: n/a
Release Notes: n/a
https://github.com/envoyproxy/envoy-mobile/issues/176

### CODE REVIEW

**Comment by alyssawilk** [comment]
tagging @tyxia for first pass

**Review by tyxia — APPROVED** [review]
LGTM, Thanks

**Comment by alyssawilk** [comment]
oh argh, merged instead of autosubmit.  @RyanTheOptimist PTAL and will address comments TBR?

File: source/extensions/tracers/xray/daemon_broker.cc [hunk 1]
```diff
@@ -32,7 +32,7 @@ DaemonBrokerImpl::DaemonBrokerImpl(const std::string& daemon_endpoint)
           Network::Utility::parseInternetAddressAndPortNoThrow(daemon_endpoint, false /*v6only*/)),
       io_handle_(Network::ioHandleForAddr(Network::Socket::Type::Datagram, address_, {})) {
   if (address_ == nullptr) {
-    throwEnvoyExceptionOrPanic(absl::StrCat("malformed IP address: ", daemon_endpoint));
+    throw EnvoyException(absl::StrCat("malformed IP address: ", daemon_endpoint));
```
Comment (RyanTheOptimist) [inline source/extensions/tracers/xray/daemon_broker.cc:35]: Out of curiosity, why this change?

File: source/common/upstream/upstream_impl.cc [hunk 2]
```diff
@@ -1793,7 +1793,7 @@ ClusterImplBase::resolveProtoAddress(const envoy::config::core::v3::Address& add
   absl::Status resolve_status;
   TRY_ASSERT_MAIN_THREAD {
     auto address_or_error = Network::Address::resolveProtoAddress(address);
-    if (address_or_error.value()) {
+    if (address_or_error.status().ok()) {
```
Comment (RyanTheOptimist) [inline source/common/upstream/upstream_impl.cc:1796]: Is this a drive-by cleanup, or is it somehow part of this change?

File: source/common/listener_manager/filter_chain_manager_impl.cc [hunk 3]
```diff
@@ -28,7 +28,7 @@ namespace FilterChain {
 // when no IP matcher is configured.
 Network::Address::InstanceConstSharedPtr fakeAddress() {
   CONSTRUCT_ON_FIRST_USE(Network::Address::InstanceConstSharedPtr,
-                         Network::Utility::parseInternetAddress("255.255.255.255"));
+                         Network::Utility::parseInternetAddressNoThrow("255.255.255.255"));
```
Comment (RyanTheOptimist) [inline source/common/listener_manager/filter_chain_manager_impl.cc:31]: Is this the only (non-test) change which actually eliminates a potential throw? It looks like all the other call sites switch from the throwing version to NoThrow but then throw if null.

**Review by RyanTheOptimist — COMMENTED** [review]
Sweet! Thanks for doing this. Just a few questions.

File: source/extensions/tracers/xray/daemon_broker.cc (hunk 1 above)
Comment (alyssawilk) [reply source/extensions/tracers/xray/daemon_broker.cc:35]: prior address cleanup used the Envoy Mobile macro when Envoy Mobile doesn't pull this extension
